src/modules/Builder/features/ActivityItems/ActivityItems.tsx

```diff
@@ -78,20 +86,32 @@ export const ActivityItems = () => {
 
       return {
         ...prevState,
-        [getItemKey(itemToDuplicate)]: numberToInsert,
+        [getEntityKey(itemToDuplicate)]: numberToInsert,
       };
     });
   };
 
-  const handleModalClose = () => {
+  const handleRemoveModalClose = () => {
     setItemIdToDelete('');
   };
 
-  const handleModalSubmit = () => {
+  const handleRemoveModalSubmit = () => {
     if (itemIdToDelete === activeItemId) setActiveItemId('');
+    if (conditionalLogicForItemToDelete?.length) {
+      const conditionalLogicKeysToRemove = conditionalLogicForItemToDelete.map(
+        (condition: ConditionalLogic) => getEntityKey(condition),
+      );
+      setValue(
+        `${fieldName}.conditionalLogic`,
+        conditionalLogic?.filter(
+          (conditionalLogic: ConditionalLogic) =>
+            !conditionalLogicKeysToRemove.includes(getEntityKey(conditionalLogic)),
```

Comment:
This peace of code is about removing matched conditions, not about making unique values.

Reply:
It can be reached with 
![image](https://github.com/ChildMindInstitute/mindlogger-admin/assets/116631639/cacaf88c-a23e-441a-9feb-89838455b153)
but up to y , y can keep it))
My solution provides better time complexity, but incr space compl))